cmd/talosctl/cmd/talos/root.go

```diff
@@ -123,3 +127,92 @@ var Commands []*cobra.Command
 func addCommand(cmd *cobra.Command) {
 	Commands = append(Commands, cmd)
 }
+
+// completeResource represents tab complete options for `ls` and `ls *` commands.
+func completePathFromNode(inputPath string) []string {
+	pathToSearch := inputPath
+
+	// If the pathToSearch is empty, use root '/'
+	if pathToSearch == "" {
+		pathToSearch = "/"
+	}
+
+	var paths map[string]struct{}
+
+	// search up one level to find possible completions
+	if pathToSearch != "/" && !strings.HasSuffix(pathToSearch, "/") {
+		splits := strings.Split(pathToSearch, "/")
+		// we need a trailing slash to search for items in a directory
+		pathToSearch = strings.Join(splits[:len(splits)-1], "/") + "/"
```

Comment:
we could also do `strings.LastIndex` (https://pkg.go.dev/strings#LastIndex) to find the index of the last `/` and use the string before it as `pathToSearch`

not that it matters, but we don't do splits which are not needed actually